integration-test/src/test/java/org/sagebionetworks/ITDownloadListControllerTest.java

```diff
@@ -90,6 +95,42 @@ public void testAddFilesToDownloadList() throws Exception {
 				.setNumberOfFilesAdded(1L);
 		assertEquals(expected, response);
 	}
+	
+	@Test
+	public void testDownloadFilesFromDownloadListWithWarehouseQuery() throws Exception {
+		FileEntity file = setupFileEntity();
+		assertEquals(
+				synapse.addFilesToDownloadList(new AddBatchOfFilesToDownloadListRequest()
+						.setBatchToAdd(Arrays.asList(new DownloadListItem().setFileEntityId(file.getId())))),
+				new AddBatchOfFilesToDownloadListResponse().setNumberOfFilesAdded(1L));
+
+		// call under test
+		DownloadListPackageResponse response = (DownloadListPackageResponse) AsyncJobHelper
+				.assertAysncJobResult(synapse, AsynchJobType.DownloadPackageList,
+						new DownloadListPackageRequest().setZipFileName("test.zip"), body -> {
+							assertTrue(body instanceof DownloadListPackageResponse);
+							DownloadListPackageResponse r = (DownloadListPackageResponse) body;
+							assertNotNull(r.getResultFileHandleId());
+						}, MAX_WAIT_MS, MAX_RETIES)
+				.getResponse();
+
+		Thread.sleep(10_000);
+		
+		Instant now = Instant.now();
+
+		String query = String.format(
+				"select count(*) from processedaccessrecord p join filedownloadrecords f on p.session_id = f.session_id where"
```

Comment:
You can join on the record_date of the processed access records so that we do not end up scanning too much data